scala/lambdas/ingest-mapper/src/main/scala/uk/gov/nationalarchives/ingestmapper/MetadataService.scala

```diff
@@ -88,7 +88,26 @@ class MetadataService(s3: DAS3Client[IO], discoveryService: DiscoveryService) {
           Stream.evals {
             topLevelIdsToDepartmentSeries.map { itemToDepartmentSeries =>
               val parentPaths = getParentPaths(json, itemToDepartmentSeries)
-              val updatedJson = jsonArr.map { metadataEntry =>
+              val childCountMap = parentPaths.groupBy(_._2).view.mapValues(_.size).toMap
+
+              def addChildCount(obj: Obj) = {
+                val values = obj.value.toMap
+                val id = values("id").str
+                val parentPath = values.get("parentPath").map(parentPath => s"${parentPath.str}/").getOrElse("")
+                val objId = s"$parentPath$id"
+                val childCount = childCountMap.getOrElse(objId, 1)
```

Comment:
Is the reason why it's get or else 1 because we're dealing with folders that must have at least 1 child? If so, do you think:

1. It's worth adding a comment or changing the name of this method to `addChildCountToFolder`?
2. Adding a test for this?

Reply:
This case is specifically for the department folder. That doesn't have a parent so isn't in the map. We're processing each top level department one at a time which means that each one will only have a single series. So if it's not in the map, it's a department folder which has 1 child.

I've added a comment to explain. The method name is fine as is and the `return the correct values from the lambda and upload the correct number of files and content to S3` is covering this path already.